internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -71,25 +94,61 @@ func walkFn(source, dest string, recursive bool, reader afero.Fs, upload UploadF
 		if !ok {
 			return nil
 		}
-		file, err := reader.Open(path)
+
+		hash := sha256.New()
+		buf := &bytes.Buffer{}
+		file, err := u.FS.Open(path)
 		if err != nil {
-			return fmt.Errorf("open file on path %q: %w", path, err)
+			return fmt.Errorf("open %q: %w", path, err)
 		}
 		defer file.Close()
-		fileRel, err := filepath.Rel(source, path)
+
+		_, err = io.Copy(io.MultiWriter(buf, hash), file)
 		if err != nil {
-			return fmt.Errorf("get relative path for %q against %q: %w", path, source, err)
+			return fmt.Errorf("copy %q: %w", path, err)
```

Comment:
should we use `file`?

Reply:
wdym? to get the path? `file` only has `Name()` 😔

Reply:
yeah but the error comes from `_, err = io.Copy(io.MultiWriter(buf, hash), file)` so I would assume `file` should be included in the error msg? Or concatenate them

Reply:
oh i see. file is a `afero.File`, so kinda hard to include in the error message 🤔